### ISSUE TITLE
`indentguide` plugin is not working on master version of lite-xl

### ISSUE THREAD

**Issue by dheisom** [lite-xl/lite-xl-plugins#40]
I have got this error while I am trying to run my lite-xl after the install of indentguide

```text
Error: /home/dheisom/.config/lite-xl/plugins/indentguide.lua:42: attempt to call a nil value (method 'subpixel_scale')
stack traceback:
	/home/dheisom/.local/share/lite-xl/core/docview.lua:427: in upvalue 'DocView_draw'
	...e/dheisom/.local/share/lite-xl/plugins/lintplus/init.lua:505: in upvalue 'draw'
	/home/dheisom/.config/lite-xl/plugins/motiontrail.lua:28: in function 'core.docview.draw'
	/home/dheisom/.local/share/lite-xl/core/commandview.lua:285: in function 'core.commandview.draw'
	/home/dheisom/.local/share/lite-xl/core/node.lua:566: in function 'core.node.draw'
	/home/dheisom/.local/share/lite-xl/core/node.lua:30: in function 'core.node.propagate'
	/home/dheisom/.local/share/lite-xl/core/node.lua:571: in function 'core.node.draw'
	/home/dheisom/.local/share/lite-xl/core/node.lua:31: in function 'core.node.propagate'
	/home/dheisom/.local/share/lite-xl/core/node.lua:571: in function 'core.node.draw'
	/home/dheisom/.local/share/lite-xl/core/node.lua:31: in function 'core.node.propagate'
	...	(skipping 1 levels)
	/home/dheisom/.local/share/lite-xl/core/node.lua:571: in function 'core.node.draw'
	/home/dheisom/.local/share/lite-xl/core/rootview.lua:406: in upvalue 'draw'
	/home/dheisom/.local/share/lite-xl/plugins/autocomplete.lua:523: in upvalue 'root_view_draw'
	/home/dheisom/.local/share/lite-xl/plugins/lsp/listbox.lua:289: in upvalue 'root_view_draw'
	/home/dheisom/.local/share/lite-xl/plugins/contextmenu.lua:32: in upvalue 'root_view_draw'
	/home/dheisom/.local/share/lite-xl/plugins/treeview.lua:454: in function 'core.rootview.draw'
	/home/dheisom/.local/share/lite-xl/core/init.lua:1561: in function 'core.step'
	/home/dheisom/.local/share/lite-xl/core/init.lua:1604: in function 'core.run'
	(...tail calls...)
	[string "local core..."]:9: in function <[string "local core..."]:2>
	[C]: in function 'xpcall'
	[string "local core..."]:2: in main chunk
```

Edit: I am using the master version of lite-xl

**Comment by Guldoman** [comment]
This is known as `master` has some breaking API changes.
You can find a version compatible with master in #1 (don't change the first line `-- mod-version:2 -- lite-xl 2.0` tho, as we've not yet bumped the API version in `master`).

**Comment by jgmdev** [comment]
A further corrected version based on PR #33

```lua
-- mod-version:2 -- lite-xl 2.0
local style = require "core.style"
local config = require "core.config"
local DocView = require "core.docview"


local function get_line_spaces(doc, idx, dir)
  local _, indent_size = doc:get_indent_info()
  local text = doc.lines[idx]
  if not text then
    return 0
  end
  local s, e = text:find("^%s*")
  if e == #text then
    return get_line_spaces(doc, idx + dir, dir)
  end
  local n = 0
  for i = s, e do
    n = n + (text:byte(i) == 9 and indent_size or 1)
  end
  return n
end


local function get_line_indent_guide_spaces(doc, idx)
  if doc.lines[idx]:find("^%s*\n") then
    return math.max(
      get_line_spaces(doc, idx - 1, -1),
      get_line_spaces(doc, idx + 1,  1))
  end
  return get_line_spaces(doc, idx)
end


local draw_line_text = DocView.draw_line_text

function DocView:draw_line_text(idx, x, y)
  local spaces = get_line_indent_guide_spaces(self.doc, idx)
  local _, indent_size = self.doc:get_indent_info()
  local w = math.ceil(1 * SCALE)
  local h = self:get_line_height()
  local sspaces = ""
  local font = self:get_font()
  for _ = 0, spaces - 1, indent_size do
    local color = style.guide or style.selection
    local sw = font:get_width(sspaces)
    renderer.draw_rect(x + sw, y, w, h, color)
    sspaces = sspaces .. (' '):rep(indent_size)
  end
  draw_line_text(self, idx, x, y)
end
```
